File: fstBookstore/fst_bookstore_backend/src/main/java/com/bookstore/service/UserService.java
package com.bookstore.service;

import com.bookstore.entity.User;
import java.util.List;

public interface UserService {
    User getUser(int userId);
    void forbidUser(int userId, boolean state);
    List<User> getUsers();
    User checkUser(String name);
    void saveUser(String username, String password, String address, List<String> emails);
}

File: bookstore_frontend/src/services/UserService.js
import {postRequest, postRequest_noReturn} from "../utils/ajax";

export const getUser = (userId, callback) => {
    const url = "/getUser";
    const data = {userId: userId};
    postRequest(url, data, callback);
}

export const forbidUser = (id, state) => {
    const url = "/forbidUser";
    const data = {userId: id, state: state};
    postRequest_noReturn(url, data);
}

export const getUsers = (callback) => {
    const url = "/getUsers";
    postRequest(url, null, callback);
}

export const saveUser = (username, password, address, email) => {
    const url = "/saveUser";
    const data = {
        username: username,
        password: password,
        address: address,
        emails: email
    };
    postRequest_noReturn(url, data);
}